### ISSUE TITLE
💅 Plugins can't analyze bare imports

### ISSUE THREAD

**Issue by tajo** [biomejs/biome#6573]
### Environment information

```bash
Tested with Biome `2.0.5` but created a reproduction directly in this project. PR attached.
```

### Rule name

Grit Plugin

### Playground link

https://github.com/biomejs/biome/pull/6572

### Expected result

It's not possible to pattern match bare imports with a Grit plugin:

```js
import "main.css";
```

with

```
`import $source`
```

Created a failing test: https://github.com/biomejs/biome/pull/6572

It works with the CLI:

```
grit apply bareImport.grit --dry-run bareImport.ts
bareImport.ts
     1  import "main.css";

Processed 1 files and found 1 matches
```

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct
